### PR TITLE
Update docs for grain-lang/grain#208

### LINKED ISSUE #64: Update docs for List with new functions
Functions added are from https://github.com/grain-lang/grain/issues/144

### PR DESCRIPTION
**Pull request by phated**
Relies on #63 

Closes #64

### CODE REVIEW

File: src/stdlib/array.md [hunk 1]
```diff
@@ -89,27 +89,35 @@ Produces a copy of the input array.
...
 
 `Array.map(array, fn)` produces a new array by calling `fn` on each element of the input array.
 
+### Array.**mapi**
+
+```grain
+mapi : ((a, Number) -> b, Array<a>) -> Array<b>
+```
+
+`Array.mapi(fn, array)` produces a new array by calling `fn` on each element of the input array, along with the index of the element.
```
Comment (ospencer) [inline src/stdlib/array.md:119]:
```suggestion
`Array.mapi(fn, array)` produces a new array by calling `fn` on each element of the input array along with its index.
```

File: src/stdlib/array.md [hunk 2]
```diff
@@ -125,3 +133,11 @@ fromList : List<a> -> Array<a>
...
 
 Converts the input list to an array.
+
+### Array.**contains**
+
+```grain
+contains : (a, Array<a>) -> Bool
+```
+
+Checks if the item is an element of the input array. Uses the generic `==` equality operator.
```
Comment (ospencer) [inline src/stdlib/array.md:143]:
```suggestion
Checks if the item is an element of the input array. Uses the generic `==` structural equality operator.
```

File: src/stdlib/list.md [hunk 3]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
 
 Evaluates the given function for each item in the list. The given function is called with the element and its index in the list.
+
+### List.**filter**
+
+```grain
+filter: (a -> Bool, List<a>) -> List<a>
+```
+
+`List.filter(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `false`, the value will be removed from the new list.
```
Comment (ospencer) [inline src/stdlib/list.md:141]:
```suggestion
`List.filter(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `false`, the value will not be included the new list.
```

File: src/stdlib/list.md [hunk 4]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+### List.**reject**
+
+```grain
+reject: (a -> Bool, List<a>) -> List<a>
+```
+
+`List.reject(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `true`, the value will be removed from the new list.
```
Comment (ospencer) [inline src/stdlib/list.md:149]:
```suggestion
`List.reject(fn, list)` produces a new list by calling `fn` on each element of the list. If the `fn` returns `true`, the value will not be included in the new list.
```

File: src/stdlib/list.md [hunk 5]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+Attempts to get the first element from the list. Fails if the list is empty.
+
+### List.**tail**
+
+```grain
+tail : List<a> -> List<a>
+```
+
+Attempts to get all elements in a list except the first element. Fails if the list is empty.
```
Comment (ospencer) [inline src/stdlib/list.md:165]:
```suggestion
Attempts to get all elements in a list except the first element. Fails if the list has one or fewer elements.
```

File: src/stdlib/list.md [hunk 6]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+Attempts to insert a value into the input list at the given index. Fails if the index is out-of-bounds.
+
+### List.**count**
+
+```grain
+count : (a -> Bool, List<a>) -> Number
+```
+
+Counts the number of elements in a list satisfy the condition.
```
Comment (ospencer) [inline src/stdlib/list.md:197]:
```suggestion
Counts the number of elements in a list that satisfy the given condition.
```

File: src/stdlib/list.md [hunk 7]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+### List.**part**
+
+```grain
+part : (Number, List<a>) -> (List<a>, List<a>)
+```
+
+Splits a list into two lists: the head of `count` elements, and the remaining elements. Fails if the count is out-of-bounds.
```
Comment (ospencer) [inline src/stdlib/list.md:205]:
```suggestion
Splits a list into two lists: a list containing the first `count` elements, and a list containing the remaining elements. Fails if the count is out-of-bounds.
```

File: src/stdlib/list.md [hunk 8]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+### List.**rotate**
+
+```grain
+rotate : (Number, List<a>) -> List<a>
+```
+
+Produces a new list where the `count` elements are moved to the end of the list.
```
Comment (ospencer) [inline src/stdlib/list.md:213]:
```suggestion
Produces a new list where the `count` elements are moved to the end of the list. If a negative count is provided, moves the last `-count` elements to the beginning of the list.
```

File: src/stdlib/list.md [hunk 9]
```diff
@@ -107,3 +131,91 @@ forEachi : ((a, Number) -> Void, List<a>) -> Void
...
+
+### List.**unique**
+
+```grain
+unique : List<a> -> List<a>
+```
+
+Produces a new list with any duplicates removed. Uses the generic `==` equality operator.
```
Comment (ospencer) [inline src/stdlib/list.md:221]:
```suggestion
Produces a new list with any duplicates removed. Uses the generic `==` structural equality operator.
```

File: src/stdlib/pervasives.md [hunk 10]
```diff
@@ -68,6 +68,22 @@ data Array<a>
...
+
+Check that two values are equal. This checks for structural equality, so it also works for comparing things like tuples and lists.
+
+### Pervasives.**(!=)**
+
+Check that two values are **not** equal. This checks for structural equality, so it also works for comparing things like tuples and lists.
+
+### Pervasives.**(eq)**
+
+Checks that two values are physically equal. Use this operator if you don't need or want structural equality.
```
Comment (ospencer) [inline src/stdlib/pervasives.md:85]: My bad for not catching this in the other PR. We already have this, and it's like Python's `is`.

```suggestion
### Pervasives.**(is)**

Checks that two values are physically equal. Use this operator if you don't need or want structural equality.
```

File: src/stdlib/list.md (hunk 5 above)
Comment (ospencer) [reply src/stdlib/list.md:165]: My brain stopped working when I wrote this—disregard.

File: src/stdlib/pervasives.md (hunk 10 above)
Comment (phated) [reply src/stdlib/pervasives.md:85]: Hah, I was staring directly at the `as` method and typed `eq`. My bad and good catch.

**Pull request merged**